eth-bytecode-db/eth-bytecode-db/src/verification/db.rs

```diff
@@ -131,21 +111,116 @@ async fn insert_source_files(
                 .context("insert into \"files\"")?,
             }
         };
+        result.push(file);
+    }
+
+    Ok(result)
+}
+
+async fn insert_source_details(
+    txn: &DatabaseTransaction,
+    source: types::Source,
+    file_models: &[files::Model],
+) -> Result<(sources::Model, bool), anyhow::Error> {
+    let abi = match source.abi {
+        None => None,
+        Some(abi) => serde_json::from_str(&abi).context("deserialize abi")?,
+    };
+
+    let file_ids = {
+        let mut file_ids = file_models.iter().map(|file| file.id).collect::<Vec<_>>();
+        file_ids.sort();
+        file_ids.dedup();
+        file_ids
+    };
+    // Results in `SELECT md5(array_agg(x)::text) FROM (VALUES (1), (2), (3)) t(x).
+    // Without t() `array_agg(x)::text` results in {(1),(2),(3)} instead of {1,2,3}.
+    let file_ids_hash_query = format!(
+        "SELECT md5(array_agg(x)::text)::uuid FROM (VALUES {}) t(x)",
```

Comment:
You should use something like Statement::from_str_and_values instead of format

Reply:
Wait what why do you use sql to calculate hash(?)?

Reply:
Updated to use `Statement::from_str_and_values` though